File: dev-docs/snippets/documents/approve-access-request.swift
import JsBaoClient

// Approve a pending access request (owner/admin only). Swift takes an untyped
// params dict and returns an untyped `[String: Any]`.
func approveAccessRequest(
  client: JsBaoClient, documentId: String, requestId: String
) async throws {
  // #region example
  let result = try await client.documents.approveAccessRequest(
    documentId: documentId,
    requestId: requestId,
    params: ["permission": "reader"]
  )
  // #endregion example
  _ = result
}

File: dev-docs/snippets/documents/approve-access-request.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Approve a pending access request (owner/admin only).
export async function approveAccessRequest(
  client: JsBaoClient,
  documentId: string,
  requestId: string,
) {
  // #region example
  const result = await client.documents.approveAccessRequest(
    documentId,
    requestId,
    { permission: "reader" },
  );
  // #endregion example
  return result;
}
